examples/graphs/node2vec_movielens.py

```diff
@@ -0,0 +1,563 @@
+            weights.append(graph[current][neighbor]["weight"])
+        else:
+            # Control the probability to move forward.
+            weights.append(graph[current][neighbor]["weight"] / q)
+
+    # Compute the probabilities of visiting each neighbor.
+    weight_sum = sum(weights)
+    probabilities = [weight / weight_sum for weight in weights]
+    # Probabilistically select a neighbor to visit.
+    next = np.random.choice(neighbors, size=1, p=probabilities)[0]
+    return next
+
+
+def randm_walk(graph, num_walks, num_steps, p, q):
+    walks = []
+    nodes = list(graph.nodes())
+    # Perform multiple iterations of the random walk.
+    for walk_iteration in range(num_walks):
+        random.shuffle(nodes)
+
+        for node in tqdm(
+            nodes,
+            position=0,
+            leave=True,
+            desc=f"Random walks iteration {walk_iteration + 1} of {num_walks}",
+        ):
+            # Start the walk with a random node from the graph.
+            walk = [node]
+            # Randomly walk for num_steps.
+            while len(walk) < num_steps:
+                current = walk[-1]
+                previous = walk[-2] if len(walk) > 1 else None
+                # Compute the next node to visit.
+                next = next_step(graph, previous, current, p, q)
+                walk.append(next)
+            # Replace node ids (movie ids) in the walk with token ids.
+            walk = [vocabulary_lookup[token] for token in walk]
+            # Add the walk to the generated sequence.
+            walks.append(walk)
+
+    return walks
+
+
+"""
+## Generate Training Data using Random Walk
+You can explore different configurations of `p` and `q` to different results of
+related movies.
+"""
+
+p = 1
+q = 1
+num_walks = 5
+num_steps = 10
+walks = randm_walk(movies_graph, num_walks, num_steps, p, q)
+
+print("Number of walks generated:", len(walks))
+
+"""
+## Generate Positive and Negative Examples
+
+To train a skip-gram model, we use the generated walks to create positive and
+negative training examples. Each example includes the following features:
+1. `target`: A movie in a walk sequence.
+2. `context`: Another movie in a walk sequence.
+3. `weight`: How many times these two movies occured in walk sequences.
+4. `label`: The label is 1 if these two movies are samples from the walk sequences,
+otherwise (i.e., if randomly sampled) the label is 0.
+"""
+
+"""
+### Generate examples
+"""
+
+
+def generate_examples(sequences, window_size, num_negative_smaples, vocabulary_size):
+
+    example_weights = defaultdict(int)
+
+    # Iterate over all sequences (walks).
+    for sequence in tqdm(
+        sequences,
+        position=0,
+        leave=True,
+        desc=f"Generating postive and negative examples",
+    ):
+
+        # Generate positive and negative skip-gram pairs for a sequence (walk).
+        pairs, labels = keras.preprocessing.sequence.skipgrams(
```

Comment:
`skipgrams` is undocumented, because it is slated for deprecation and removal (we think it is too niche of a utility, although we can be convinced otherwise). I would recommend including your own implementation of it in the example. How many lines of code is the example so far?

Reply:
It seems to me that its documentation is OK: https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/sequence/skipgrams

In addition, it is used to this TF example: https://www.tensorflow.org/tutorials/text/word2vec

I think it is a useful utility and it would be great to keep it (both in this example and in Keras :) )

LMKWDYT

Reply:
Acknowledged, the usage in the word2vec example is a good argument. We'll move it to `utils`, presumably.